mobile-app/app/screens/AppNavigator/screens/Loans/VaultDetail/VaultDetailScreenV2.tsx

```diff
@@ -0,0 +1,375 @@
+import { VaultDetailLoansRow } from "./components/VaultDetailLoansRow";
+import { VaultDetailCollateralsRow } from "./components/VaultDetailCollateralsRow";
+import { LoanParamList } from "../LoansNavigator";
+
+type Props = StackScreenProps<LoanParamList, "VaultDetailScreen">;
+
+export function VaultDetailScreenV2({ route, navigation }: Props): JSX.Element {
+  const { isLight } = useThemeContext();
+  const { vaultId } = route.params;
+  const [vault, setVault] = useState<LoanVault>();
+  const canUseOperations = useLoanOperations(vault?.state);
+  const dispatch = useAppDispatch();
+  const client = useWhaleApiClient();
+  const blockCount = useSelector((state: RootState) => state.block.count);
+  const isFocused = useIsFocused();
+  const { collateralTokens } = useCollateralTokenList();
+
+  const vaults = useSelector((state: RootState) => vaultsSelector(state.loans));
+  const loanTokens = useSelector((state: RootState) =>
+    loanTokensSelector(state.loans)
+  );
+
+  const inLiquidation = vault?.state === LoanVaultState.IN_LIQUIDATION;
+  const totalLoanAmount = new BigNumber(
+    inLiquidation || vault === undefined ? 0 : vault?.loanValue
+  );
+  const totalCollateralValue = new BigNumber(
+    inLiquidation || vault === undefined ? 0 : vault?.collateralValue
+  );
+  const minColRatio = new BigNumber(
+    vault === undefined ? 0 : vault?.loanScheme.minColRatio
+  );
+  const vaultState = useVaultStatus(
+    vault?.state,
+    new BigNumber(
+      inLiquidation || vault === undefined ? 0 : vault?.informativeRatio
+    ),
+    minColRatio,
+    totalLoanAmount,
+    totalCollateralValue
+  );
+  const collateralAmounts =
+    inLiquidation || vault === undefined ? [] : vault.collateralAmounts;
+  const loanAmounts =
+    inLiquidation || vault === undefined ? [] : vault.loanAmounts;
+  const nextCollateralizationRatio = useNextCollateralizationRatio(
+    collateralAmounts,
+    loanAmounts
+  );
+  const maxLoanAmount = useMaxLoan({
+    totalCollateralValue: totalCollateralValue,
+    collateralAmounts: collateralAmounts,
+    existingLoanValue: totalLoanAmount,
+    minColRatio: minColRatio,
+  });
+
+  const [snapPoints, setSnapPoints] = useState({
+    ios: ["75%"],
+    android: ["70%"],
+  });
+  const [bottomSheetScreen, setBottomSheetScreen] = useState<
+    BottomSheetNavScreen[]
+  >([]);
+  const getBottomSheetHeader = (label: string) => {
+    return {
+      headerStatusBarHeight: 2,
+      headerTitle: "",
+      headerBackTitleVisible: false,
+      headerStyle: tailwind("rounded-t-xl-v2 border-b-0", {
+        "bg-mono-light-v2-100": isLight,
+        "bg-mono-dark-v2-100": !isLight,
+      }),
+      header: () => (
+        <BottomSheetTokenListHeader
+          headerLabel={label}
+          onCloseButtonPress={dismissModal}
+        />
+      ),
+    };
+  };
+  const {
+    bottomSheetRef,
+    containerRef,
+    dismissModal,
+    expandModal,
+    isModalDisplayed,
+  } = useBottomSheet();
```

Comment:
we can use the `bottomSheetScreen` and `setBottomSheetScreen` from this hook too, then dont have create to local state

Reply:
similarly for `snapPoints` and `setSnapPoints`

Reply:
done